beacon-chain/blockchain/execution_engine.go

```diff
@@ -218,14 +218,22 @@ func (s *Service) notifyNewPayload(ctx context.Context, preStateVersion int,
 	var lastValidHash []byte
 	if blk.Version() >= version.Deneb {
 		var versionedHashes []common.Hash
+		var requests *enginev1.ExecutionRequests
+
+		if blk.Version() >= version.Electra {
+			requests, err = blk.Block().Body().ExecutionRequests()
+			if err != nil {
+				return false, errors.Wrap(err, "could not get execution requests")
+			}
+		}
 		versionedHashes, err = kzgCommitmentsToVersionedHashes(blk.Block().Body())
 		if err != nil {
 			return false, errors.Wrap(err, "could not get versioned hashes to feed the engine")
 		}
 		pr := common.Hash(blk.Block().ParentRoot())
-		lastValidHash, err = s.cfg.ExecutionEngineCaller.NewPayload(ctx, payload, versionedHashes, &pr)
+		lastValidHash, err = s.cfg.ExecutionEngineCaller.NewPayload(ctx, payload, versionedHashes, &pr, requests)
 	} else {
-		lastValidHash, err = s.cfg.ExecutionEngineCaller.NewPayload(ctx, payload, []common.Hash{}, &common.Hash{} /*empty version hashes and root before Deneb*/)
+		lastValidHash, err = s.cfg.ExecutionEngineCaller.NewPayload(ctx, payload, []common.Hash{}, &common.Hash{} /*empty version hashes and root before Deneb*/, nil)
```

Comment:
While you're editing this it would be nice to remove this inline comment on a method argument.

I also think using zero values and removing nesting could improve legibility:
```
var versionedHashes []common.Hash
var parentRoot *common.Hash
var requests *enginev1.ExecutionRequests
if blk.Version() >= version.Deneb {
	versionedHashes, err = kzgCommitmentsToVersionedHashes(blk.Block().Body())
	if err != nil {
		return false, errors.Wrap(err, "could not get versioned hashes to feed the engine")
	}
	prh := common.Hash(blk.Block().ParentRoot())
	parentRoot = &prh
}
if blk.Version() >= version.Electra {
	requests, err = blk.Block().Body().ExecutionRequests()
	if err != nil {
		return false, errors.Wrap(err, "could not get execution requests")
	}
}

lastValidHash, err = s.cfg.ExecutionEngineCaller.NewPayload(ctx, payload, versionedHashes, parentRoot, requests)
```

Reply:
nice improvement updating